pkg/codegen/python/importer.go

```diff
@@ -59,6 +59,13 @@ type PackageInfo struct {
 	PyProject struct {
 		Enabled bool `json:"enabled,omitempty"`
 	} `json:"pyproject,omitempty"`
+
+	// Specifies if TypedDict based types should be generated.
+	// Allowed values are the following:
+	// TODO[pulumi/pulumi/16375]: Flip default to side-by-side
+	// - "none" (default): do not generate TypedDict based types
+	// - "side-by-side": generate TypedDict based types side-by-side with Args classes.
+	TypedDictArgs string `json:"typedDictArgs,omitempty"`
```

Comment:
@lunaris pointed out that it might make more sense to name this `Args` with values `classes`, `dicts`, `both` (or maybe `classes-and-dicts` instead of `both`)